core/src/main/java/hudson/scm/SCM.java

```diff
@@ -368,7 +398,7 @@ public SCMRevisionState _calcRevisionsFromBuild(AbstractBuild<?, ?> build, Launc
      *      this exception should be simply propagated all the way up.
      * @since 1.568
      */
-    public PollingResult compareRemoteRevisionWith(@Nonnull Job<?,?> project, @Nullable Launcher launcher, @Nullable FilePath workspace, @Nonnull TaskListener listener, @Nonnull SCMRevisionState baseline) throws IOException, InterruptedException {
+    public PollingResult compareRemoteRevisionWith(@Nonnull Job<?,?> project, @Nullable Launcher launcher, @Nonnull FilePath workspace, @Nonnull TaskListener listener, @Nonnull SCMRevisionState baseline) throws IOException, InterruptedException {
```

Comment:
you are not forcing the developer to override `compareRemoteRevisionWith(Job, Launcher, TaskListener, SCMRevisionState)` though - so you can not make that guarantee